pkg/operators/combiner/combiner.go

```diff
@@ -173,7 +186,13 @@ func (o *combinerOperatorInstance) forwardData(
 	if config.interval == 0 {
 		// Define a maximum waiting time for data from all targets
 		// TODO: Make it configurable?
-		timeout := time.NewTimer(5 * time.Second)
+		defaultTimeout := 5 * time.Second
+		if config.count == 0 {
+			// We are only handling array data sources here
+			// If interval and count are both 0, wait till the user stops the gadget
+			defaultTimeout = 24 * 365 * time.Hour
+		}
+		timeout := time.NewTimer(defaultTimeout)
```

Comment:
What about letting `c` be nil instead of using one year? (waiting on nil blocks forever, so the result is the same in practice)

Reply:
Even better!